mantidproject/mantid
# Add missing pytest files and hook to detect them

**Description of work.**
- Adds a pre-commit hook from https://github.com/DavidFair/pre-commit-hooks/tree/Add_pytest_checking
- Add the various missing files from the project into their CMake lists

**To test:**
- Code Review https://github.com/mantidproject/pre-commit-hooks/pull/5
- Try running pre-commit using `pre-commit run cmake-missing-pytest-files --verbose --all-files` it should pass
- Remove (a) random file(s) from CMake lists, then re-run check they are correctly detected as missing

If you are happy with the changes:
- Approve the PR in our hooks repo
- Ping me, and I'll update the URL from my fork to the Mantid version
- Then this will be ready for merge

There is no associated issue - I thought we created an issue for missing tests, but cannot find it?

*This does not require release notes* because it's not user visible



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
